src/session/session_dhandle.c

```diff
@@ -357,6 +363,8 @@ __wt_session_get_btree(WT_SESSION_IMPL *session,
 	bucket = __wt_hash_city64(uri, strlen(uri)) % WT_HASH_ARRAY_SIZE;
 	SLIST_FOREACH(dhandle_cache, &session->dhhash[bucket], hashl) {
 		dhandle = dhandle_cache->dhandle;
+		if (F_ISSET(dhandle, WT_DHANDLE_DEAD))
+			continue;
```

Comment:
This test looks wrong to me -- if we do this, and then continue to open the file, won't we end up with two references in the session cache to the same file?